cfme/tests/automate/test_service_automate.py

```diff
@@ -455,3 +459,185 @@ def test_retire_vm_now(setup_provider, full_template_vm, new_user):
             full_template_vm.retire()
             assert full_template_vm.wait_for_vm_state_change(desired_state="retired", timeout=720,
                                                              from_details=True)
+
+
+@pytest.fixture
+def custom_prov_data(request, provisioning):
+    prov_data = {
+        "catalog": {'vm_name': random_name()},
+        "environment": {'automatic_placement': True},
+        "network": {'vlan': partial_match(provisioning['vlan'])}
+    }
+    value = request.param
+    prov_data.update(value)
+    prov_data['catalog']['catalog_name'] = {'name': provisioning["template"]}
+    return prov_data
+
+
+@pytest.fixture(scope='function')
+def domain_setup(domain):
+    """This fixture used to setup the domain structure"""
+    # Instantiating class 'QuotaMethods'
+    klass = (
+        domain.parent.instantiate(name="ManageIQ")
+        .namespaces.instantiate(name="System")
+        .namespaces.instantiate(name="CommonMethods")
+        .classes.instantiate(name="QuotaMethods")
+    )
+
+    # Instantiating method 'requested' and copying it to new domain
+    klass.methods.instantiate(name="requested").copy_to(domain.name)
+
+    method = (
+        domain.parent.instantiate(name=f"{domain.name}")
+        .namespaces.instantiate(name="System")
+        .namespaces.instantiate(name="CommonMethods")
+        .classes.instantiate(name="QuotaMethods")
+        .methods.instantiate(name="requested")
+    )
+    yield
+    method.delete_if_exists()
+
+
+@pytest.fixture(scope='module')
+def tenants_setup(appliance):
+    """This fixture creates two parent tenants"""
+    parent_tenant = appliance.collections.tenants.create(
```

Comment:
If it is not necessary to create these tenants via UI, creating them via REST would be faster. You can then instantiate a UI entity.

Reply:
No. According to me, It is better to go with UI.